### PR TITLE
Recipe for embedding reduction [Videos]

### PR DESCRIPTION
**Pull request by plon-Susk7**
This PR is related to issue #410 

1. Added notebook example for embedding reduction using dimension_reduction operator.
2.  Used Hugging face dataset  "sayakpaul/ucf101-subset" which has 10 classes, as example.
3. Used vid_vec_rep_clip operator to extract embeddings.

### CODE REVIEW

**Comment by aatmanvaidya** [comment]
@plon-Susk7 great work! 
just a few things

1. can we add some more 1-2 lines descriptions in the markdown -- you have the headings, just one two more lines explaining the process - like you are extracting embedding's using CLIP etc etc
2. I am guessing that you have to download the `huggingface hub`, `matplotlib` libraries? where are you downloading them? in the `.venv`? what if we download them in the notebook? what I mean is, what if the first cell is something like this
```sh
!pip install huggingface-hub
!pip install matplotlib
!pip install datasets
```
This way we make sure the user has to only run the notebook and they should not worry about fixing package install issue. 
3. The final plot looks great, is there a chance that the plot is a bit more spatial, like right now the thumbnails overlap a lot
4. Also, how many videos are there in the dataset?

File: src/notebooks/03_plot_tsne_videos.ipynb [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+   "execution_count": 4,
+   "id": "aa8c9e2d-066e-43b7-add8-2b5d5293d1c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Downloading and extracting \n",
+    "\n",
+    "dataset_name = \"UCF101_subset/train\"\n",
+    "hf_dataset_identifier = \"sayakpaul/ucf101-subset\"\n",
+    "filename = \"UCF101_subset.tar.gz\"\n",
```
Comment (aatmanvaidya) [inline src/notebooks/03_plot_tsne_videos.ipynb:87]: does the `tar.gz` file gets deleted in the end?

Comment (plon-Susk7) [reply src/notebooks/03_plot_tsne_videos.ipynb]: This tar.gz files get stores in python cache.
'/home/python/.cache/huggingface/hub/datasets--sayakpaul--ucf101-subset/snapshots/b9984b8d2a95e4a1879e1b071e9433858d0bc24a/UCF101_subset.tar.gz'.
I don't think we need to specifically remove these, but I may be wrong. What do you think?

**Comment by plon-Susk7** [comment]
> This way we make sure the user has to only run the notebook and they should not worry about fixing package install issue. 3. The final plot looks great, is there a chance that the plot is a bit more spatial, like right now the thumbnails overlap a lot 4. Also, how many videos are there in the dataset?

There are 10 classes in the dataset, so I took 5 from each. In total there are 50 videos.

**Comment by aatmanvaidya** [comment]
@plon-Susk7 this looks great, merging the PR now

can you also do one small thing - on the issue - https://github.com/tattle-made/feluda/issues/410, can you write detailed instructions on how to download the jupyter notebook in the `.venv` and then exec into the docker container and run it.
I will add those instructions to the wiki

**Pull request merged**